### PR TITLE
[WIP] smartcard: fix listing when no reader is available

### PR DESCRIPTION
**Pull request by hardening**
This patch fixes enumeration of readers when nothing is connected as in #7214.

### CODE REVIEW

File: channels/smartcard/client/smartcard_pack.c [hunk 1]
```diff
@@ -205,17 +205,14 @@ static BOOL smartcard_ndr_pointer_write(wStream* s, UINT32* index, DWORD length)
 	return TRUE;
 }
 
-static LONG smartcard_ndr_write(wStream* s, const BYTE* data, UINT32 size, UINT32 elementSize,
+static LONG smartcard_ndr_write(wStream* s, const BYTE* data, UINT32 count, UINT32 elementSize,
```
Comment (akallabeth) [inline channels/smartcard/client/smartcard_pack.c:208]: unrelated changes

**Review by akallabeth — CHANGES_REQUESTED** [review]
* The changes in `smartcard_ListReadersA_Call` and `smartcard_ListReadersW_Call` differ but should not (e.g. cchReaders = 0;)
* `smartcard_ndr_write` changes are unrelated

**Comment by hardening** [comment]
Ok, comparing with packets sent by mstsc I figured that there's differences, from what I see it's a problem with padding. So I'm gonna clean up this PR to only fix the empty list of readers and then I'll push more fixes.
